fault/tester.py

```diff
@@ -53,3 +57,31 @@ def serialize(self):
     def compile_and_run(self, target="verilator", **kwargs):
         target_inst = self.make_target(target, **kwargs)
         target_inst.run()
+
+    def __check_interfaces_match(self, old_circuit, new_circuit):
```

Comment:
Seems that this function can be moved outside of the class. Perhaps in common or in this file outside of the class. Maybe rename `check_circuit_is_subset` or something, because we only care that all ports in the old_circuit are in the new circuit.

It maybe also possible to make this late-binding. We may not have a strict subset but still be able to 'copy' because all the accessed ports exist in the new circuit. Depends on how tightly we want to enforce things.